src/view/RepeatedFragment.js

```diff
@@ -210,6 +210,10 @@ export default class RepeatedFragment {
 			if ( newIndex === -1 ) return;
 
 			const fragment = this.iterations[ oldIndex ];
+
+			// guard against null fragment - #2005 (feels like treating the symptom, not the cause?)
+			if ( fragment == null ) return;
```

Comment:
I'm on board with using `foo == null` represent a check against a primitive or multi-type value for not existing (`null` or `undefined`). When I see that  `== null` then I know that's exactly the condition of that variable - it could exist but have a possible falsey value like `0`.

However, in this case `fragment` is either an object or not, so `if ( !fragment )` should be sufficient and it communicates that expectation.